### PR TITLE
Add API Tests to Core

### PR DESCRIPTION
**Pull request by ludydoo**
This PR sets up some basic API tests to be run
It ensures that we can do basic operations such as
- Creating, Getting, Listing databases
- Creating, Getting, Listing folders
- Creating, Getting, Listing forms
- Creating, Getting, Listing records

We use `embedded-postgres` to run an actual in-process database. Since `go test` does not run all tests on the same process, we must synchronize setup/teardown of the database using a file lock.

### CODE REVIEW

File: pkg/server/public/tests/form_test.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		}, {
+			name: "With Required Reference Field",
+			fields: tu.Fields(
+				tu.AReferenceField(otherForm, tu.FieldName("My Field"), tu.FieldRequired(true)),
+			),
+		}, {
+			name: "With Key Reference Field",
+			fields: tu.Fields(
+				tu.AReferenceField(otherForm, tu.FieldName("My Field"), tu.FieldRequired(true), tu.FieldKey(true)),
+			),
```
Comment (nilueps) [inline pkg/server/public/tests/form_test.go:139]: Add cases for broken inputs. i.e.: missing required fields, wrong types, etc

File: pkg/sqlmanager/schema_test.go [hunk 2]
```diff
@@ -1,50 +1,65 @@
...
+	exitVal := m.Run()
+	done()
+	os.Exit(exitVal)
+}
+
 func (s *Suite) SetupSuite() {
 
-	pg := embeddedpostgres.NewDatabase()
-	if err := pg.Start(); !assert.NoError(s.T(), err) {
+	sqlDb, err := gorm.Open(postgres.Open("host=localhost port=15432 user=postgres password=postgres dbname=postgres sslmode=disable"))
```
Comment (nilueps) [inline pkg/sqlmanager/schema_test.go:45]: can we set the port dynamically?

**Review by nilueps — COMMENTED** [review]
Clean & Snappy

File: pkg/server/public/tests/form_test.go (hunk 1 above)
Comment (ludydoo) [reply pkg/server/public/tests/form_test.go:139]: Yeah... this part was just to test the rest api, make sure that we can read and write without breaking anything. Def. would need more work in the future, more cases, etc.

File: pkg/sqlmanager/schema_test.go (hunk 2 above)
Comment (ludydoo) [reply pkg/sqlmanager/schema_test.go:45]: Yeah we could add an env var like TEST_POSTGRES_PORT=... and read from it, with a default

**Comment by ludydoo** [comment]
@nilueps  lgtm ?

File: pkg/sqlmanager/schema_test.go (hunk 2 above)
Comment (nilueps) [reply pkg/sqlmanager/schema_test.go:45]: Yea

**Comment by nilueps** [comment]
there's a conflict

**Pull request merged**